unumpy/_multimethods.py

```diff
@@ -825,12 +826,14 @@ def broadcast_to(array, shape, subok=False):
 
 
 def _normalize_axis(ndim, axis, allow_repeats=False):
-    if isinstance(axis, int):
+    if isinstance(axis, numbers.Number):
         axis = (axis,)
```

Comment:
This is incorrect, there may be functions that only accept a single integer as an axis rather than a tuple, so it may be best to make integers the base case, and call the function recursively for sequences.

Reply:
So the recursive part would exist in the multimethod and not in `_normalize_axis`?

Reply:
Never mind my last question, it doesn't make sense. 😅

Reply:
Something like this:

```python
def _normalize_axis(axis, ...):
    if isinstance(axis, Sequence):
        # Make sure no nesting
        out = tuple(_normalize_axis(a, ...) for a in axis)
        # Repeats etc. handled here
        return out
    # Integers handled here
```

Reply:
It's been a while since I've last implemented a recursive function. Here's what I came up with:

```python
def _normalize_axis(ndim, axis, allow_repeats=False):
    if isinstance(axis, numbers.Number):
        axis_pos = ndim + operator.index(axis) if axis < 0 else operator.index(axis)
    
        if axis_pos >= ndim or axis_pos < 0:
            raise ValueError(
                "Axis %s is out of bounds for array of dimension %s." % (axis, ndim)
            )
        
        return axis_pos
    
    if isinstance(axis, tuple):
        if len(axis) == 1:
            return _normalize_axis(ndim, axis[0])
        else:
            axes = [_normalize_axis(ndim, axis[0])] + [_normalize_axis(ndim, axis[1:])]

            if not allow_repeats and (len(axes) != len(set(axes))):
                raise ValueError("Repeated axis.")

            return tuple(axes)
```

Reply:
Multiple levels of recursion don't make sense, if you look at my template, it explains how to do it.